src/main/java/io/kokuwa/keycloak/metrics/event/MetricsEventListener.java

```diff
@@ -2,22 +2,25 @@
 
 import java.util.Optional;
 
+
 import org.jboss.logging.Logger;
+import org.keycloak.Config;
 import org.keycloak.events.Event;
 import org.keycloak.events.EventListenerProvider;
 import org.keycloak.events.admin.AdminEvent;
 import org.keycloak.models.KeycloakContext;
 import org.keycloak.models.KeycloakSession;
 import org.keycloak.models.RealmModel;
+import org.keycloak.provider.EnvironmentDependentProviderFactory;
 
 import io.micrometer.core.instrument.Metrics;
-
+import io.kokuwa.keycloak.metrics.CommunityProfiles;
 /**
  * Listener for {@link Event} and {@link AdminEvent}.
  *
  * @author Stephan Schnabel
  */
-public class MetricsEventListener implements EventListenerProvider, AutoCloseable {
+public class MetricsEventListener implements EventListenerProvider, AutoCloseable, EnvironmentDependentProviderFactory {
```

Comment:
This doesn't go in the provider. It's for the ProviderFactory implementation

Reply:
Thanks for the headsup